dagshub/data_engine/client/data_client.py

```diff
@@ -134,6 +166,16 @@ def _datasource_query(self, datasource: Datasource, include_metadata: bool, limi
         return self._exec(q, params)["datasourceQuery"]
 
     def update_metadata(self, datasource: Datasource, entries: List[DatapointMetadataUpdateEntry]):
+        """
+        Create a new datasource using the provided datasource state.
```

Comment:
I'm pretty sure this isn't what this function does.

Reply:
Yeah. I think i copied another functions description here.